### ISSUE TITLE
[APP SUBMITTED]: viewlog() got an unexpected keyword argument 'limit'

### ISSUE THREAD

**Issue by OmgImAlexis** [pymedusa/SickRage#457]
### INFO
Python Version: **2.7.11 (default, Dec 15 2015, 16:46:19) [GCC 4.8.4]**
Operating System: **Linux-4.1.18-unRAID-x86_64-with-Ubuntu-14.04-trusty**
Locale: UTF-8
Branch: **master**
Commit: PyMedusa/SickRage@55af59244f853c38812832343f2209fec59bc8de
No Log available with ERRORS:
### ERROR
```
Thread-14 :: [55af592] Failed doing webui callback: Traceback (most recent call last):
  File "/app/sickrage/sickbeard/webserve.py", line 276, in async_call
    result = function(**kwargs)
TypeError: viewlog() got an unexpected keyword argument 'limit'

```
---
_STAFF NOTIFIED_: @SickRage/support @SickRage/moderators

**Comment by OmgImAlexis** [comment]
Webserve should sent all params to each method and then the method should just discard all unneeded ones instead of erroring out like this.

**Comment by fernandog** [comment]
You got this error because you manually edited the URL. viewlog works fine. Not an issue

**Comment by OmgImAlexis** [comment]
@fernandog there's no reason adding a param to a URL should cause an error like this. It's a bug.

**Comment by fernandog** [comment]
LOL i give up

**Comment by labrys** [comment]
@fernandog Calling an invalid url on a website shouldn't generate an error for the server is the point.  It should raise a 404 if the result isn't reachable.

@OmgImAlexis The reason this generates an error is so the developer can decide how to handle the URL.  For example should it silently ignore the extra params or should it return a 404 as that resource with those params is unavailable.  Cases can be made for both, but I'm of the opinion it should be a 404.

**Comment by OmgImAlexis** [comment]
@labrys going off the HTTP status codes we should only be returning an "error" if it's missing something.
For example sending a 422 if there's missing params, if there's extra ones we should just ignore them.

@fernandog i'd rather know all the bugs we have to fix and be able to get this repo to be as close to 100% bug free as we can instead of ignoring things like this as we seemed to in the old repo. Being able to show that our application only errors out when there's actually an error is better than seeing it error out because of something so trivial as this.

**Comment by labrys** [comment]
@OmgImAlexis Technically as per [RFC 7231](https://tools.ietf.org/html/rfc7231) 400 would be the correct status code, but this type of ambiguity as to which code to use for which situation is exactly why it just raises an error, its up to the developer to catch that error and decide what response would be the most appropriate.  IMO the "easy" fix would be to return a 404 on any invalid request for any reason, and then work towards providing correct return statuses for situations as they are encountered.  Ideally the best response for this particular situation, IMO, is to return a 400 with a message to the effect of "Received unknown parameter 'limit'", potentially even with a "suggestion" link below such as "Retry request without the unknown parameter: <URL WITHOUT PARAMETER>"

**Comment by ratoaq2** [comment]
Fixed in #890 
Extra parameters are silently ignored
